pragma-node/src/infra/repositories/onchain_repository.rs

```diff
@@ -477,3 +475,169 @@ pub async fn get_publishers_with_components(
 
     Ok(publishers_response)
 }
+
+// --- onchain OHLC ---
+
+pub async fn get_ohlc(
+    ohlc_data: &mut Vec<OHLCEntry>,
+    pool: &Pool,
+    network: Network,
+    pair_id: String,
+    interval: Interval,
+    data_to_retrieve: i64,
+) -> Result<(), InfraError> {
+    let now = Utc::now();
+    let aligned_current_timestamp = interval.align_timestamp(now);
+    let start_timestamp = if data_to_retrieve > 1 {
+        aligned_current_timestamp
+            - Duration::minutes(interval.to_minutes() * (data_to_retrieve * 10))
+    } else {
+        aligned_current_timestamp
+    };
+
+    let entries = get_entries_from_timestamp(pool, network, &pair_id, start_timestamp).await?;
+    update_ohlc_data(
+        ohlc_data,
+        entries,
+        interval,
+        now,
+        start_timestamp,
+        data_to_retrieve == 1,
+    );
+
+    Ok(())
+}
+
+async fn get_entries_from_timestamp(
+    pool: &Pool,
+    network: Network,
+    pair_id: &str,
+    start_timestamp: DateTime<Utc>,
+) -> Result<Vec<SpotEntry>, InfraError> {
+    let raw_sql = format!(
+        r#"
+            SELECT
+                *
+            FROM
+                {table_name}
+            WHERE
+                pair_id = '{pair_id}'
+                AND timestamp >= '{start_timestamp}'
+            ORDER BY
+                timestamp
+            ASC
+        "#,
+        table_name = get_table_name(network, DataType::SpotEntry),
+        pair_id = pair_id,
+        start_timestamp = start_timestamp
+    );
+
+    let conn = pool.get().await.map_err(adapt_infra_error)?;
+    let entries: Vec<SpotEntry> = conn
+        .interact(move |conn| diesel::sql_query(raw_sql).load::<SpotEntry>(conn))
+        .await
+        .map_err(adapt_infra_error)?
+        .map_err(adapt_infra_error)?;
+    Ok(entries)
+}
+
+fn update_ohlc_data(
```

Comment:
rustdoc + how efficient is this for bigger intervals ?
I think an application layer cache would be quite important here especially when computing past ohlc

Reply:
Tried for our biggest interval and didn't notice any issue - I would say that adding a cache adds quite the complexity overhead so that's more something that I would implement later when proven useful (i.e. if requests are very slow for our usage)